rfc/2020/alertmanager-integration/christyjacob4.md

```diff
@@ -0,0 +1,503 @@
+| `May 25th -  June 1st` | Define Alertmanager Charts and Alertmanager RBAC yaml files <br> Enable Alertmanager as an addOn option using the default rules and receivers |
+| `June 1st -  June 8th` | Define functions for reading the rules and config files and unmarshalling them from file paths passed in `config.yaml` <br> Blog Post #3 |
+| `June 8th -  June 15th` |  Add support for `config.yaml` file to be passed to the `addon-config` flag. <br>  Perform extensive testing of the `config.yaml `functionality <br> |
+| | *Phase #1 Evaluation Starts `( June 15th )`* |
+| `June 15th - June 22nd` |  Write Integration Tests and define golden files for Alertmanager addOn functionality.<br> Add a check in `linkerd check` to ensure that Alertmanager has been successfully installed and correctly configured.<br> Blog Post #4 |
+| `June 22nd -  June 29th` | Tests to ensure that `linkerd check` command is functional and update the golden files <br> Configure Alertmanager for High Availability Mode |
+| `June 29th -  July 6th` | Extensive Testing of all the aforementioned functionality <br> Integration Test cases for Alertmanager in  High Availability mode <br> Blog Post #5 |
+| `July 6th -  July 13th` |  Generate Service Profiles for apps with swagger spec, record their per-route metrics and define default Prometheus rules for them  |
+| | *Phase #2 Evaluation Starts `( July 13th )`* |
+| `July 13th -  July 20th` | Generate Service Profiles for apps with protobuf spec, record their per-route metrics and define default Prometheus rules for them <br> Blog Post #6 |
+| `July 20th -  July 27th` | Auto generate Service Profiles for apps without swagger spec or protobuf spec and record their per-route metrics and define default Prometheus rules for them |
+| `July 27th -  August 3rd` | Other Service Profile integration related action items  <br> Blog Post #7 |
+| `August 3rd -  August 10th` | Integration Tests for Service Profile Integration <br> Buffer Time for any action items that may not have been completed |
+| | *Final Evaluation Starts `( August 10th )`* |
+| `August 10th -  August 17th` |   Discuss changes for allowing write access from Linkerd Dashboard <br> Design UI components to support Add/Modify/Delete Prometheus Rules and Alertmanager receivers <br> Blog Post #8 |
+
+
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+- I understand that `linkerd profile` can be used to generate Service Profiles based on protobufs, swagger specs, or by using tap to observe the web service. I would like to understand the kind of integration that is expected as part of [Out of the box alerts #1726](https://github.com/linkerd/linkerd2/issues/1726) . Does it mean that the `linkerd profile` command must be automatically run if the user installs Alertmanager, and the generated Service Profile needs to be part of the Alertmanager rules? 
```

Comment:
Service profiles are usually created by service/application owners, which usually happens **after** Linkerd is installed. You won't be able to create service profiles before the Linkerd  control plane is fully installed.

The only thing that Prometheus and Alertmanager care about is the alert rules.  Can we install Prometheus and Alertmanager with some default non-service profile alerts (based on the [3 golden metrics](https://linkerd.io/2/features/telemetry/)), and provide a way for users to add alerts which are based on route metrics after the service profiles are installed?

Reply:
Yep, what are even installing service profiles for? because we wouldn't even have the application installed (in most cases) before Linkerd is installed (and add-on install goes with control-plane). 
Also all our default alerts would be for the golden metrics of control-plane components. right?

I think, Once after the users installs the application and has service profiles done, they can just go and add more rules (w.r.t serviceprofiles) and perform a upgrade right?